src/expire.c

```diff
@@ -261,7 +261,7 @@ void activeExpireCycle(int type) {
                         /* Get the next entry now since this entry may get
                          * deleted. */
                         dictEntry *e = de;
-                        de = de->next;
+                        de = dictGetNext(de);
```

Comment:
i think we need to convert this code to work with an iterator rather than directly on the tables and next pointer.
otherwise dict is still not properly abstracted.

Reply:
I've converted this to use dictScan.